languages/rust/oso/src/oso.rs

```diff
@@ -69,10 +70,11 @@ impl Oso {
 
     fn check_inline_queries(&self) -> crate::Result<()> {
         while let Some(q) = self.inner.next_inline_query(false) {
+            let location = q.source_info();
             let query = Query::new(q, self.host.clone());
             match query.collect::<crate::Result<Vec<_>>>() {
                 Ok(v) if !v.is_empty() => continue,
-                Ok(_) => return lazy_error!("inline query result was false"),
+                Ok(_) => return Err(OsoError::InlineQueryFailedError { location }),
                 Err(e) => return lazy_error!("error in inline query: {}", e),
```

Comment:
Should this error be updated too to the new type?

Reply:
darn git hell - yeah, somehow that goes backed out

Reply:
don't understand what you're asking for here.  this IS the new type?

Reply:
On L78 we still use `lazy_error`, but it's representing a failure in an inline query

Reply:
that's actually a different error - one is failure, the other is that it threw
I think it's better to leave them separate - I suppose we could add a new variant for it.

Reply:
I agree that it's fine to leave it — the two errors are distinct. Whether we want to create a special new error type for errors (as distinct from failures) in inline queries feels outside the scope of this PR.

Reply:
Got it. I just looked into the other adapters.  When an error occurs in an inline query we just raise it directly without wrapping it in anything.

We also have a consistent way to attach source information to all errors in other adapters: https://github.com/osohq/oso/blob/83884a26ddb36d1c828187216d5aa713790ed469/languages/python/oso/polar/exceptions.py#L13

For now, I think we should also include the location information on L78 so that an error in an inline query can be debugged by our users (otherwise it will have no stack trace, like failed inline queries before this PR).

I'm going to write down another task in Notion to add stack traces to all errors in the Rust adapter (similar to Python).

Reply:
Er - never mind.  We don't need to do anything because the stack trace will be printed when the error is formatted if it has one.

https://github.com/osohq/oso/blob/83884a26ddb36d1c828187216d5aa713790ed469/polar-core/src/error.rs#L285